From 0ed6fdae073077a02d9d180b1b65d51912434e00 Mon Sep 17 00:00:00 2001
From: Lars Glud <larshg@gmail.com>
Date: Thu, 5 Feb 2026 20:48:53 +0100
Subject: [PATCH] Use same QT vesion as VTK. (#6406)

* Use same QT vesion as VTK.

* Move the setting of WITH_QT to AUTO to else block, if VTK is not used/found.
---
 CMakeLists.txt | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9d887e5fb19..168c3aba1ca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -423,9 +423,19 @@ if(WITH_VTK)
   endif()
 endif()
 
-# VTK can depend on Qt and search for its required version, so search after so we can overwrite Qt5_FOUND/Qt6_FOUND if the version we require is not found.
-set(WITH_QT "AUTO" CACHE STRING "Build QT Front-End (AUTO|YES|QT6|QT5|NO)")
+# If VTK has found a Qt version, default to the same version
+if(VTK_QT_VERSION STREQUAL "5")
+  message(STATUS "VTK has found Qt5 - defaulting to same version.")
+  set(WITH_QT "QT5" CACHE STRING "Build QT Front-End (AUTO|YES|QT6|QT5|NO)")
+elseif(VTK_QT_VERSION STREQUAL "6")
+  message(STATUS "VTK has found Qt6 - defaulting to same version.")
+  set(WITH_QT "QT6" CACHE STRING "Build QT Front-End (AUTO|YES|QT6|QT5|NO)")
+else()
+  # VTK can depend on Qt and search for its required version, so search after so we can overwrite Qt5_FOUND/Qt6_FOUND if the version we require is not found.
+  set(WITH_QT "AUTO" CACHE STRING "Build QT Front-End (AUTO|YES|QT6|QT5|NO)")
+endif()
 set_property(CACHE WITH_QT PROPERTY STRINGS "AUTO" "YES" "QT6" "QT5" "NO")
+
 if(WITH_QT)
   include("${PCL_SOURCE_DIR}/cmake/pcl_find_qt.cmake")
 endif()
